qiskit/algorithms/linear_solvers/observables/absolute_average.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The absolute value of the average of the components of the vector solution to the
+linear systems."""
+from typing import Union, Optional, List
+import numpy as np
+
+from qiskit.aqua.operators import PauliSumOp
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.opflow import I, Z, TensoredOp
+from .linear_system_observable import LinearSystemObservable
+
+
+class AbsoluteAverage(LinearSystemObservable):
+    r"""A class for the absolute value of the average of the components of the vector solution
+    to the linear systems.
+    For a vector :math:`x=(x_1,...,x_N)`, the average is defined as
+    :math:`\abs{\frac{1}{N}\sum_{i-1}^{N}x_i}`."""
+    def observable(self, num_qubits: int) -> Union[PauliSumOp, List[PauliSumOp]]:
+        """The observable operator.
+
+        Args:
+            num_qubits: The number of qubits on which the observable will be applied.
+
+        Returns:
+            The observable as a sum of Pauli strings.
+        """
+        zero_op = ((I + Z) / 2)
+        return TensoredOp(num_qubits * [zero_op])
+
+    def post_rotation(self, num_qubits: int) -> Union[QuantumCircuit, List[QuantumCircuit]]:
+        """The observable circuit.
+
+        Args:
+            num_qubits: The number of qubits on which the observable will be applied.
+
+        Returns:
+            The observable as a QuantumCircuit.
+        """
+        qc = QuantumCircuit(num_qubits)
+        qc.h(qc.qubits)
+        return qc
+
+    def post_processing(self, solution: Union[float, List[float]],
+                        num_qubits: int,
+                        constant: Optional[float] = 1) -> float:
```

Comment:
This should just be `constant: float = 1` as mentioned elsewhere in regards of default versus Optional.

Can we also perhaps give it a better name - reading the docstring how how `scaling`, `scale`, or `scale_factor` for example

Reply:
removed Optional and changed `constant` to `scaling`.